guix-science-nonfree/packages/linux.scm

```diff
@@ -86,3 +86,12 @@
 		   (invoke "make" "install"))))))
     (synopsis
      "Intel PSM2 communication library, with NVIDIA GPU Direct support")))
+
+;; libfabric in itself doesn't have specific CUDA support but
+;; GPUDirect support requires the CUDA enabled version of its MSP2
+;; dependency.
+(define-public libfabric-cuda
+  (package/inherit libfabric
+    (name (string-append (package-name libfabric) "-cuda"))
+    (inputs (modify-inputs (package-inputs libfabric)
+              (replace "psm2" psm2-cuda)))))
```

Comment:
Looks like something's not working as expected; quoth the build log:

```
checking cuda_runtime.h usability... no
checking cuda_runtime.h presence... no
checking for cuda_runtime.h... no
checking whether cuMemGetHandleForAddressRange is declared... no
checking whether CU_DEVICE_ATTRIBUTE_DMA_BUF_SUPPORTED is declared... no
checking whether CU_MEM_RANGE_HANDLE_TYPE_DMA_BUF_FD is declared... no
```

And also:

```
$ guix gc --references /gnu/store/zky8f9g4mbid7ji2qna9nm5wdhvavawj-libfabric-cuda-1.22.0
/gnu/store/3lfhjyimi6fwnjzk198rc37s6pwngiim-libnl-3.5.0
/gnu/store/930nwsiysdvy2x5zv1sf6v7ym75z8ayk-gcc-11.3.0-lib
/gnu/store/9hibvfbxgcg25qryy57p0k7zrl0w0spy-rdma-core-48.0
/gnu/store/gsjczqir1wbz8p770zndrpw4rnppmxi3-glibc-2.35
/gnu/store/zky8f9g4mbid7ji2qna9nm5wdhvavawj-libfabric-cuda-1.22.0
```

Reply:
Fixed in version 6.